docassemble/ALToolbox/data/questions/al_income.yml

```diff
@@ -89,4 +89,602 @@ data: !!omap
+question: |
+  Tell us about this asset
+fields:
+  - What type of asset is it?: x.source
+    code: |
+      asset_source_list
+    exclude: |
+      ['vehicle']
+  - How often do you get this income?: x.times_per_year
+    input type: radio
+    code: |
+      times_per_year_list
+    datatype: integer
+  - Interest or other income from the asset: x.value
+    datatype: currency
+  - Who owns this?: x.owner
+    required: False
+---
+# UNIQUE FOR ALAssetList
+---
+generic object: ALAssetList
+code: |
+  x[i].source
+  x[i].market_value
+  x[i].complete = True
+---
+id: number of assets
+generic object: ALAssetList
+question: |
+  How many assets do you have?
+fields:
+  - no label: x.target_number
+    datatype: integer
+---
+id: info for each asset
+generic object: ALAssetList
+question: |
+  About your ${ x[i].source } asset
+fields:
+  - Current market value: x[i].market_value
+    datatype: currency
+    required: False
+  - Balance (for a financial account, may be identical to market value): x[i].balance
+    datatype: currency
+    required: False
+  - How much does this asset earn you (in interest, dividends or other regular payments to you)?: x[i].value
+    datatype: currency
+    required: False
+  - How often do you get money for this asset?: x[i].times_per_year
+    input type: radio
+    code: |
+      times_per_year_list
+    datatype: integer
+    required: False
+---
+# UNIQUE FOR ALVehicle
+---
+generic object: ALVehicle
+code: |
+  # .source is automatically 'vehicle'
+  x.market_value
+  x.complete = True
+---
+comment: |
+  Notes:
+
+  A Vehicle is a child class of Asset, and so it has attributes like value
+  (representing income earned), times_per_year, balance, and market_value, but
+  for most vehicle owners only the `balance` and `market_value` attributes from
+  ALAsset will be meaningful.
+id: one vehicle info
+generic object: ALVehicle
+question: |
+  Tell us about your vehicle
+fields:
+  - Year: x.year
+    input type: combobox
+    code: |
+      recent_years()
+  - Make: x.make
+    hint: e.g, Ford, Honda
+  - Model: x.model
+    hint: e.g., Fusion, Civic
+  - Market value: x.market_value
+    datatype: currency
+  - Loan balance: x.balance
+    datatype: currency
+    required: False
+  - Income earned: x.value
+    datatype: currency
+    required: False
+  - How often is the income earned?: x.times_per_year
+    datatype: integer
+    code: times_per_year_list
+  - Who owns this?: x.owner
+---
+# UNIQUE FOR ALVehicleList
+---
+# Has to have its own. Otherwise it's overridden by ALAssetList.
+generic object: ALVehicleList
+code: |
+  # .source is automatically 'vehicle'
+  x[i].market_value
+  x[i].complete = True
+---
+id: how many vehicles
+generic object: ALVehicleList
+question: |
+  How many vehicles do you have?
+fields:
+  - no label: x.target_number
+    datatype: integer
+---
+comment: |
+  Notes:
+  
+  * Do not try to use the `total()` method with an ALVehicle. ALVehicleList.
```

Comment:
Not sure about this comment. the `.total()` method should return the total income from all vehicles, which is generally speaking 0 but it's not semantically wrong. Also both comments are weirdly placed--probably belong in the class definition or docstring

Reply:
That said there a number of similar comments and it's fine to leave most of them in--just suggest removing this one incorrect one for now

Reply:
I'll remove that one.

These comments were originally in the `subquestion` of the block, because they used to be in the demo, not a production interview; I didn't want to move them too far.